mapbox/services/matrix.py

```diff
@@ -0,0 +1,41 @@
+"""Matrix API V1"""
+
+import re
+
+from mapbox.encoding import encode_waypoints
+from mapbox.errors import InvalidProfileError
+from mapbox.services.base import Service
+
+
+class DirectionsMatrix(Service):
+    """Access to the Matrix API V1"""
+
+    api_name = 'directions-matrix'
+    api_version = 'v1'
+
+    valid_profiles = [
+        'mapbox/driving', 'mapbox/cycling', 'mapbox/walking',
+        'mapbox/driving-traffic']
+
+    @property
+    def baseuri(self):
+        return 'https://{0}/{1}/{2}'.format(
+            self.host, self.api_name, self.api_version)
+
+    def _validate_profile(self, profile):
+        parts = re.split(r'[\/\.]', profile)
+        if len(parts) == 1:
+            parts = ['mapbox'] + parts
+        profile = '/'.join(parts)
```

Comment:
We've got some inconsistency in the APIs: `mapbox.driving` in some places and `mapbox/driving` in the new API. To help users, we can support each, as well as upgrading bare `driving` to `mapbox/driving`.

Reply:
I took a slightly different approach in the Direction v5 PR: https://github.com/mapbox/mapbox-sdk-py/blob/ee78696f1924ba48b23f1d951f3f1213d849a357/mapbox/services/directions.py#L32-L45

Just noting that there is some overlap and we should consider pulling into a common function. I'd prefer a warning if using the old `.` style.